### PR TITLE
Merge `package:pubspec_parse`

### PR DESCRIPTION
**Pull request by mosuem**
- [x] Move and fix workflow files, labeler.yaml, and badges in the README.md
- [x] Rev the version of the package, so that pub.dev points to the correct site
- [x] Add a line to the changelog:
```
* Move to `dart-lang/tools` monorepo.
```

- [x] Add the package to the top-level readme of the monorepo:
```
| [pubspec_parse](pkgs/pubspec_parse/) | Simple package for parsing pubspec.yaml files with a type-safe API and rich error reporting. | [![package issues](https://img.shields.io/badge/package:pubspec_parse-4774bc)](https://github.com/dart-lang/tools/issues?q=is%3Aissue+is%3Aopen+label%3Apackage%3Apubspec_parse) | [![pub package](https://img.shields.io/pub/v/pubspec_parse.svg)](https://pub.dev/packages/pubspec_parse) |
```

- [ ] **Important!** Merge the PR with 'Create a merge commit' (enabling then disabling the `Allow merge commits` admin setting)
- [x] Update the auto-publishing settings on https://pub.dev/packages/pubspec_parse/admin
- [x] Add the following text to https://github.com/dart-lang/pubspec_parse/:'

```
> [!IMPORTANT]  
> This repo has moved to https://github.com/dart-lang/tools/tree/main/pkgs/pubspec_parse
```

- [ ] Publish using the autopublish workflow
- [ ] Push tags to GitHub using
```
git tag --list 'pubspec_parse*' | xargs git push origin
```

- [ ] Close open PRs in dart-lang/pubspec_parse with the following message:

```
Closing as the [dart-lang/pubspec_parse](https://github.com/dart-lang/pubspec_parse) repository is merged into the [dart-lang/tools](https://github.com/dart-lang/tools) monorepo. Please re-open this PR there!
```
      
- [ ] Transfer issues by running
```
dart run pkgs/repo_manage/bin/report.dart transfer-issues --source-repo dart-lang/pubspec_parse --target-repo dart-lang/tools --add-label package:pubspec_parse --apply-changes
```

- [ ] Archive https://github.com/dart-lang/pubspec_parse/

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Comment by spydon** [comment]
@mosuem can we land https://github.com/dart-lang/pubspec_parse/pull/137 and https://github.com/dart-lang/pubspec_parse/pull/140 before moving the package?

**Comment by devoncarew** [comment]
as an update, I did land https://github.com/dart-lang/pubspec_parse/pull/137

File: pkgs/pubspec_parse/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: pubspec_parse
+version: 1.3.1
+description: >-
+  Simple package for parsing pubspec.yaml files with a type-safe API and rich
+  error reporting.
+repository: https://github.com/dart-lang/tools/tree/main/pkgs/pubspec_parse
+topics:
```
Comment (devoncarew) [inline pkgs/pubspec_parse/pubspec.yaml:7]:
```suggestion

topics:
```

**Review by devoncarew — APPROVED** [review]
lgtm

**Comment by spydon** [comment]
> @mosuem can we land [dart-lang/pubspec_parse#137](https://github.com/dart-lang/pubspec_parse/pull/137) and [dart-lang/pubspec_parse#140](https://github.com/dart-lang/pubspec_parse/pull/140) before moving the package?

Now both are merged in the other repository. :)

**Pull request merged**